projects/memapCore/src/main/java/memap/helper/JsonExportHelper/CouplerJsonHelper.java

```diff
@@ -0,0 +1,133 @@
+package memap.helper.JsonExportHelper;
+
+import com.google.gson.annotations.Expose;
+
+import memap.messages.extension.NetworkType;
+import memap.messages.planning.CouplerMessage;
+
+/**
+ * Coupler is the class for representing Coupler objects. Example: CHP, HeatPump
+ */
+public class CouplerJsonHelper{
+
+	@Expose
+	private String name;
+	@Expose
+	private String networkTypeP; // Primary, Values: Heat or Electricity
+	@Expose
+	private String networkTypeS; // Secondary, Values: Heat or Electricity
+	@Expose
+	private double minimumPower;
+	@Expose
+	private double maximumPower;
+	@Expose
+	private double efficiencyPrimary;
+	@Expose
+	private double efficiencySecondary;
+	@Expose
+	private double cost;
+	@Expose
+	private double coEmission;
+
+	/**
+	 * Constructor for class Coupler
+	 * 
+	 * @param name                an alphanumeric string
+	 * @param networkTypeP        primary network type. A string: Heat or
+	 *                            Electricity
+	 * @param networkTypeS        secondary network type. A string: Heat or
+	 *                            Electricity
+	 * @param minimumPower        a positive double [kW]
+	 * @param maximumPower        a positive double [kW]
+	 * @param efficiencyPrimary   primary network efficiency. A positive double
+	 * @param efficiencySecondary secondary network efficiency. A negative double
+	 * @param cost                cost [EUR]
+	 * @param coEmission          CO2 Emissions measured in [g/kWh]
+	 */
+	public CouplerJsonHelper(CouplerMessage couplerMessage) {
+
+		setName(couplerMessage.name);
+		setNetworkTypeP(couplerMessage.primaryNetwork);
+		setNetworkTypeS(couplerMessage.secondaryNetwork);
+		setMinimumPower(couplerMessage.minPower);
+		setMaximumPower(couplerMessage.maxPower);
+		// TODO: Problem! Primary/Secondary Efficiency in Coupler Message!
+		setEfficiencyPrimary(couplerMessage.efficiencyHeat);
+		setEfficiencySecondary(couplerMessage.efficiencyElec);
```

Comment:
If the primary efficiency in line 55 is reserved for heat, why not setting the primary network in line 50 also to heat? The same is probably valid for electricity in lines 56 and 51.

Reply:
This is directly related to issue #155 :
I think we have to change/refactor the coupler message from efficiencyHeat to efficiencyPrimary and efficiencyElec to efficiencySecondary to account for both heat pump and CHP. 

I have the same problem in `OpcUaBuildingController.java` line 284-285

Reply:
true, this is a good idea. Only issue is that this changes different parts in the code, starting message, gui front end and json model. Maybe we can make this in one specific branch later on, not now.

Reply:
The GUI already uses the terms "efficiencyPrimary" and "efficiencySecondary" in Coupler devices. The GuiController also incorporates these terms. Thus, changing the variable names would only affect classes in memapCore.